gradle/verification-metadata.xml

```diff
@@ -2436,121 +2436,241 @@
             <sha256 value="97467f6177be7ea1c64cf7cfbdad8f8c18d472a18950106d35138dd05010a754" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.apache.lucene" name="lucene-analysis-common" version="9.4.0-snapshot-923a9f800ae">
+         <artifact name="lucene-analysis-common-9.4.0-snapshot-923a9f800ae.jar">
+            <sha256 value="163c5002f4a4115af1d83c785dac61f71d2173becaf0208fd98b110222a003d5" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.apache.lucene" name="lucene-analysis-icu" version="9.3.0">
          <artifact name="lucene-analysis-icu-9.3.0.jar">
             <sha256 value="e3af1f650877bcc438e15ea2b12dfac4d3ac12c27105cd4b000d0916730f6a74" origin="Generated by Gradle"/>
          </artifact>
       </component>
```

Comment:
I'm not familiar with this file, is it expected that it still lists 9.3.0 artifacts?

Reply:
@jpountz Thanks a good comment. I don't know. I will ping ES delivery team to confirm.
These changes are autogenerated by the command `./gradlew --write-verification-metadata sha256 precommit` that we are using instead of `./gradlew updateShas` now.

Reply:
@elastic/es-delivery We found it surprising that `verification-metadata.xml` file contains both artifacts for lucene-9.3.0 and the upgrade we are working on lucene-9.4.0.  These changes are autogenerated by the command `./gradlew --write-verification-metadata sha256 precommit` that we are asked to used.

We want to confirm with you that they are expected?

Reply:
I believe that's because `--wirte-verification-metadata` simply _appends_ to the existing file. It does not `cull` older entries. We may decide to cleanup older dependency entries at a later time.

Reply:
@mark-vieira Thanks for confirmation. For now we will proceed with this PR as it is. 
I am not sure who is responsible for cleaning older dependencies. Please let us know if our team should do at a later time.